src/vm/stackingallocator.h

```diff
@@ -215,87 +212,45 @@ class StackingAllocator
     unsigned    m_MaxAlloc;
 #endif
 
-    void Init(bool bResetInitBlock)
-    {
-        WRAPPER_NO_CONTRACT;
-
-        if (bResetInitBlock || (m_InitialBlock == NULL))
-        {
-            Clear(NULL);
-            m_FirstBlock = NULL;
-            m_FirstFree = NULL;
-            m_BytesLeft = 0;
-            m_InitialBlock = NULL;
-        }
-        else
-        {
-            m_FirstBlock = m_InitialBlock;
-            m_FirstFree = m_InitialBlock->m_Data;
-            _ASSERTE(FitsIn<unsigned>(m_InitialBlock->m_Length));
-            m_BytesLeft = static_cast<unsigned>(m_InitialBlock->m_Length);
-        }
-    }
+    void Init();
 
 #ifdef _DEBUG
-    void Validate(StackBlock *block, void* spot)
-    {
-        LIMITED_METHOD_CONTRACT;
-
-        if (!block) 
-            return;
-        Sentinal* ptr = block->m_Sentinal;
-        _ASSERTE(spot);
-        while(ptr >= spot)
-        {
-                // If this assert goes off then someone overwrote their buffer!
-                // A common candidate is PINVOKE buffer run.  To confirm look
-                // up on the stack for NDirect.* Look for the MethodDesc
-                // associated with it.  Be very suspicious if it is one that
-                // has a return string buffer!.  This usually means the end
-                // programmer did not allocate a big enough buffer before passing
-                // it to the PINVOKE method.
-            if (ptr->m_Marker1 != Sentinal::marker1Val)
-                _ASSERTE(!"Memory overrun!! May be bad buffer passed to PINVOKE. turn on logging LF_STUBS level 6 to find method");
-            ptr = ptr->m_Next;
-        }
-        block->m_Sentinal = ptr;
-    }
+    void Validate(StackBlock *block, void* spot);
 #endif
 
-    void Clear(StackBlock *ToBlock)
-    {
-        LIMITED_METHOD_CONTRACT;
-
-        StackBlock *p = m_FirstBlock;
-        StackBlock *q;
-
-        while (p != ToBlock)
-        {
-            PREFAST_ASSUME(p != NULL);
-            
-            q = p;
-            p = p->m_Next;
-
-            INDEBUG(Validate(q, q));
-
-            // we don't give the tail block back to the OS
-            // because we can get into situations where we're growing
-            // back and forth over a single seam for a tiny alloc
-            // and the perf is a disaster -- VSWhidbey #100462
-            if (m_DeferredFreeBlock != NULL)
-            {
-                delete [] (char *)m_DeferredFreeBlock;
-            }
-
-            m_DeferredFreeBlock = q;
-            m_DeferredFreeBlock->m_Next = NULL;
-        }
-    }
+    void Clear(StackBlock *ToBlock);
 
 private :
     static Checkpoint s_initialCheckpoint;
 };
 
+#define ACQUIRE_STACKING_ALLOCATOR(stackingAllocatorName)  \
```

Comment:
Can we please instead make a function for this. There are too many steps in this macro and it will make debugging annoying.

Reply:
Just saw the alloca. Never mind.